smithy-model/src/main/java/software/amazon/smithy/model/validation/NodeValidationVisitor.java

```diff
@@ -388,6 +391,32 @@ public List<ValidationEvent> memberShape(MemberShape shape) {
         return events;
     }
 
+    public List<ValidationEvent> checkNullMember(MemberShape shape) {
+        List<ValidationEvent> events = new ArrayList<>();
```

Comment:
Not sure if this is a concern for NodeValidationVisitor performance, but we could return a static empty list (`Collections.emptyList()`) unless at least one error is actually found.

Reply:
True, this could be rewritten to return `ListUtils.of` based collections instead of creating resizable lists.